### ISSUE TITLE
bonus2 bRegenPercentHP / bRegenPercentSP are completely broken.

### ISSUE THREAD

**Issue by NeutralDev** [rathena/rathena#4507]
* **rAthena Hash**: 34e5f53

* **Client Date**: 2017-06-14bRagexeRE

* **Server Mode**: pre-renewal

* **Description of Issue**: **bonus2 bRegenPercentHP & bonus2 bRegenPercentSP** don't do anything when applied to items. Apparently this problem has existed for at least 2+ years, as mentioned in [this topic with no replies](https://rathena.org/board/topic/113364-item-bonus-hp-p-ercent-regen-not-working-bonus2-bregenpercenthpnt/).
  * How to Reproduce: Just equip something with the bonus2 bRegenPercentHP and/or bonus2 bRegenPercentSP scripts.

* **Modifications that may affect results**: 
I personally fixed this myself after debugging and doing some backtracking. There are two parts to this problem, the first one can be found at **status.cpp, status_natural_heal**, at this if statement block:

https://github.com/rathena/rathena/blob/e27aa8b2d82620aa461291b293e00afb7c79e0cb/src/map/status.cpp#L14307-L14312

As you can probably see here, **sd->percent_hp_regen.value** and **sd->percent_sp_regen.value** are never used to call the pc_regen function, therefore the script will never work. This needs to be extended to include the following if statement at the bottom (or just extend the current if statement at the bottom):

```
if (sd->percent_hp_regen.value || sd->percent_sp_regen.value)
    pc_regen(sd, natural_heal_diff_tick);
```

Then after this is resolved, the percentage formula for these functions also don't work, as referenced here in **pc.cpp, pc_regen**:

https://github.com/rathena/rathena/blob/e27aa8b2d82620aa461291b293e00afb7c79e0cb/src/map/pc.cpp#L11039-L11053

These formulas for percentage don't work, since there's no **/ 100** at the end after the multiplication step. If you run these right now after the pc_regen function starts recognizing this script, it will fully heal you every single time. They should really look like this:

`hp += sd->status.max_hp * sd->percent_hp_regen.value / 100;`
`sp += sd->status.max_sp * sd->percent_sp_regen.value / 100;`

After the above modifications, these functions will start working properly.

**Comment by yashimwong** [comment]
I think you should create a pull request as you seem to already have the solution

**Comment by NeutralDev** [comment]
Will do.

**Comment by Lemongrass3110** [comment]
Fixed in bc5b147
